internal/locate/region_request.go

```diff
@@ -616,7 +616,8 @@ func (state *accessFollower) next(bo *retry.Backoffer, selector *replicaSelector
 	if selector.targetIdx < 0 {
 		leader := selector.replicas[state.leaderIdx]
 		leaderEpochStale := leader.isEpochStale()
-		leaderInvalid := leaderEpochStale || state.IsLeaderExhausted(leader)
+		leaderUnreachable := leader.store.getLivenessState() != reachable
+		leaderInvalid := leaderEpochStale || leaderUnreachable || state.IsLeaderExhausted(leader)
```

Comment:
It's better to print the states `leaderEpochStale`,  `leaderUnreachable` and `state.IsLeaderExhausted(leader)` in the logs at L622.